### ISSUE TITLE
process_priority ignored when calling _ready

### ISSUE THREAD

**Issue by Jack-023** [godotengine/godot#78783]
### Godot version

v4.0.3.stable.official [5222a99f5]

### System information

Windows 11

### Issue description

From the docs `_ready` call order can be overridden by setting `process_priority`
https://docs.godotengine.org/en/stable/tutorials/scripting/scene_tree.html#tree-order

>This can also be overridden using the process_priority node property. Nodes with a lower number are called first. For example, nodes with the priorities "0, 1, 2, 3" would be called in that order (from left to right).

I have tried setting this value both in code and in the editor and it always executes in the default order (in order of generations from the root from top to bottom)

### Steps to reproduce

I have created a scene with this structure
```
Node
|__Parent
   |___child
   |___child2
```

where parent, child and child2 all have the code 
```
func _ready() -> void:
    print("%s ready with priority %s" % [name, process_priority])
```

Here are a couple of examples with some different values to see what happens.
```
child ready with priority 1
child2 ready with priority 50
parent ready with priority 1
```
```
child ready with priority 99
child2 ready with priority 0
parent ready with priority 1
```

As you can see process_priority has no effect.

### Minimal reproduction project

All the repro steps are above but here is the project for convenience. 

[bug.zip](https://github.com/godotengine/godot/files/11890805/bug.zip)

**Comment by Sauermann** [comment]
Currently the iteration order in a `HashMap` is not sorted in any way:

https://github.com/godotengine/godot/blob/d95b516d312a9f3930a96c0fdd51cf975345f16f/scene/main/node.cpp#L220-L222

https://github.com/godotengine/godot/blob/d95b516d312a9f3930a96c0fdd51cf975345f16f/scene/main/node.h#L153

This was done for efficiency reasons in #75627. So it would probably be best to update the docs to describe the changed behavior.

**Comment by Jack-023** [comment]
The part that is actually relevant for my use case is getting the parent to initialize before the children. Is there another way to achieve this if the functionality described in the docs has been removed?

**Comment by AThousandShips** [comment]
Parent will never run `_ready` before children no matter what the specific implementation here, by design

**Comment by akien-mga** [comment]
You should use `_enter_tree` (or `_init`, if better suited) if you want to initialize the parents before the children. See the description in the Node class for details on execution order.

**Comment by kleonc** [comment]
> if the functionality described in the docs has been removed?

It hasn't been removed, AFAICT `process_priority` has never been affecting the order of the readiness. It affects the order of... processing (since it was added in #19888). The docs are incorrect. They werer changed like that in https://github.com/godotengine/godot-docs/pull/5449 (cc @skyace65, @Calinou).

So this needs to be corrected/clarified in the 4.0 docs, and updated for 4.1 according to the new behavior (https://github.com/godotengine/godot/issues/78783#issuecomment-1610838191).

---

> Parent will never run `_ready` before children no matter what the specific implementation here, by design

It can, if the child node is added after the parent is ready (so likely not when initializing but still).

**Comment by AThousandShips** [comment]
> It can, if the child node is added after the parent is ready

True, should have been clear that it won't when adding a hierarchy like loading etc., important clarification
